app/controllers/books_controller.rb

```diff
@@ -0,0 +1,9 @@
+class BooksController < ApplicationController
+  include Wor::Paginate
+  before_action :authenticate_user!
+  def show; end
```

Comment:
You need to define show method. You can read `rails routes` output and note there is something like this: `GET /books/:id`
Hint: `#show` method receives a param named id, find this object in the database, and render it.

Reply:
Done